### PR TITLE
fix(capacity_reservation): take `nemesis_grow_shrink_instance_type` into account

### PR DESCRIPTION
**Pull request by fruch**
`nemesis_grow_shrink_instance_type` was introduced, but on the capacity_reservation logic we didn't took it into account

Fixes: #8514

### Testing

- [ ]

### PR pre-checks (self review)

- [x] I added the relevant `backport` labels
- [x] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevant to this change (if needed)

### CODE REVIEW

File: sdcm/provision/aws/capacity_reservation.py [hunk 1]
```diff
@@ -33,7 +33,14 @@ class SCTCapacityReservation:
     def _get_cr_request_based_on_sct_config(params) -> Tuple[dict[str, int], int]:
         instance_counts = {}
         nemesis_node_count = params.get("nemesis_add_node_cnt") or 0
-        cluster_max_size = (params.get("cluster_target_size") or params.get("n_db_nodes")) + nemesis_node_count
+
+        cluster_max_size = (params.get("cluster_target_size") or params.get("n_db_nodes"))
+
+        if nemesis_grow_shrink_instance_type := params.get("nemesis_grow_shrink_instance_type"):
+            instance_counts[nemesis_grow_shrink_instance_type] = nemesis_node_count
```
Comment (soyacz) [inline sdcm/provision/aws/capacity_reservation.py:40]: in unrealistic scenario when `nemesis_grow_shrink_instance_type` is the same like `instance_type_db` or like monitoring/loader node type then this cause problems. 
```
instance_counts[nemesis_grow_shrink_instance_type] = instance_counts.get(nemesis_grow_shrink_instance_type, 0) + nemesis_node_count
```

**Review by soyacz — APPROVED** [review]
Besides of error for unrealistic scenario.
LGTM

File: sdcm/provision/aws/capacity_reservation.py (hunk 1 above)
Comment (fruch) [reply sdcm/provision/aws/capacity_reservation.py:40]: we can guard that in the configuration itself, i.e. forbid using the same instance type as db as part of validation ?

Comment (soyacz) [reply sdcm/provision/aws/capacity_reservation.py:40]: I think above proposal is easier and I think we don't have to restrict it then.

Comment (fruch) [reply sdcm/provision/aws/capacity_reservation.py:40]: why not use defaultdict with 0 ? and add to it as needed ?

Comment (soyacz) [reply sdcm/provision/aws/capacity_reservation.py:40]: also can be done this way.

Comment (fruch) [reply sdcm/provision/aws/capacity_reservation.py]: it's looks much better now

I've test it with the unittests, but didn't want to complex them with add this logic into them

**Pull request merged**